core/TerminologyDesignationResolver.py

```diff
@@ -0,0 +1,413 @@
+        Since they are retrieved from the FHIR resources stored by the terminology server, their values are regarded as
+        official values and thus override any existing code designation mappings already stored by the object
+        :param ui_tree_dir: Path to directory containing UI tree JSON files
+        :param value_set_dir: Path to directory containing expanded `ValueSet` resource FHIR JSON files
+        """
+        logger.debug("Loading base translations")
+        if ui_tree_dir is None:
+            ui_trees = []
+        else:
+            ui_trees = [json.load(open(os.path.join(ui_tree_dir, file_name), mode='r', encoding='UTF-8'))
+                        for file_name in os.listdir(ui_tree_dir) if file_name.endswith('.json')]
+        if value_set_dir:
+            value_sets = []
+        else:
+            value_sets = [json.load(open(os.path.join(value_set_dir, file_name), mode='r', encoding='UTF-8'))
+                          for file_name in os.listdir(value_set_dir) if file_name.endswith('.json')]
+        if not self.base_translation_mapping:
+            logger.warning("No mapping seems to exist and thus no designations can be loaded. Aborting")
+        else:
+            for ui_tree in ui_trees:
+                self.__load_base_designations_for_ui_tree(ui_tree)
+            for value_set in value_sets:
+                self.__load_base_designations_for_value_set(value_set)
+
+    def load_designations(self, folder_path: str = os.path.join("..", "example", "code_systems_translations"),
+                          update_translation_supplements=False):
+        """
+        Loads the code_system files from specified folder into memory
+        :param folder_path: folder containing translated code_systems
+        :param update_translation_supplements: specifies if supplements should be downloaded or updated from TERMINOLOGY_SERVER_ADDRESS
+        """
+        if update_translation_supplements:
+            logger.info("Downloading and Updating from the supplement registry ")
+            response_supp_registry = REQUESTS_SESSION.get(
+                url=self.server_address + "CodeSystem/fdpg-plus-translation-supplement-registry",
+                cert=(SERVER_CERTIFICATE, PRIVATE_KEY)
+            )
+            if response_supp_registry.status_code != 200:
+                logger.info("Something went wrong. Status code:" + response_supp_registry.status_code +" Expected 200")
+                logger.info("Content:"+response_supp_registry.content)
+
+            supplement_registry = response_supp_registry.json()
+            for code_system_concept in supplement_registry.get('concept'):
+                for prop in code_system_concept.get('property'):
+                    if prop.get('code') == "supplement-canonical":
+
+                        code_system_url = prop.get('valueString').split('|')[0]
+                        code_system_version = prop.get('valueString').split('|')[-1]
```

Comment:
@Frontman50 , @paulolaup  - this only works if there is a version in the url, otherwise it uses the code system url as version - fix like this or similar:

```
                     code_system_url_split = prop.get('valueString').split('|')
                        code_system_url = code_system_url_split[0]

                        if len(code_system_url_split) == 2:
                            code_system_version = prop.get('valueString').split('|')[-1]
```